src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NecessaryCanonicalEETypeNode.cs

```diff
@@ -25,6 +26,8 @@ protected override ISymbolNode GetBaseTypeNode(NodeFactory factory)
             return _type.BaseType != null ? factory.NecessaryTypeSymbol(_type.BaseType.NormalizeInstantiation()) : null;
         }
 
+        protected override FrozenRuntimeTypeNode GetFrozenRuntimeTypeNode(NodeFactory factory) => throw new NotSupportedException();
```

Comment:
Instantiations over `__Canon` are not visible to reflection and we don't need to ever access their `WritableData`. I'm making it so that we don't reserve space in the BSS section and emit a useless pointer here. See the change in `EETypeNode::.ctor` and `EETypeNode::OutputWritableData`.